### PR TITLE
Few small fixes to allow envoy to compile on ARM32

### PR DESCRIPTION
**Pull request by costinm**
Tested compilation for rasbperry pi and android (the 'endian' change is
required for android NDK - recent androids can compile with 64 bit).

Signed-off-by: Costin Manolache <costin@gmail.com>

### CODE REVIEW

**Comment by costinm** [comment]
Note that I compiled/tested using a custom cmake, not bazel.

File: source/server/options_impl.cc [hunk 1]
```diff
@@ -24,9 +24,9 @@ OptionsImpl::OptionsImpl(int argc, char** argv, const std::string& hot_restart_v
   log_levels_string += "\n[trace] and [debug] are only available on debug builds";
 
   TCLAP::CmdLine cmd("envoy", ' ', VersionInfo::version());
-  TCLAP::ValueArg<uint64_t> base_id(
+  TCLAP::ValueArg<uint32_t> base_id(
```
Comment (htuch) [inline source/server/options_impl.cc]: Why are these necessary?

Comment (costinm) [reply source/server/options_impl.cc]: I get this error with the pi cross-compiler. Not sure if there is some other fix, not an expert in c++... 

In file included from /ws/istio-master/src/tclap/include/tclap/Arg.h:54:0,
                 from /ws/istio-master/src/tclap/include/tclap/SwitchArg.h:30,
                 from /ws/istio-master/src/tclap/include/tclap/CmdLine.h:27,
                 from /ws/istio-master/envoy/source/server/options_impl.cc:13:
/ws/istio-master/src/tclap/include/tclap/ArgTraits.h: In instantiation of 'struct TCLAP::ArgTraits<long long unsigned int>':
/ws/istio-master/src/tclap/include/tclap/ValueArg.h:403:66:   required from 'void TCLAP::ValueArg<T>::_extractValue(const string&) [with T = long long unsigned int; std::string = std::basic_string<char>]'
/ws/istio-master/src/tclap/include/tclap/ValueArg.h:363:29:   required from 'bool TCLAP::ValueArg<T>::processArg(int*, std::vector<std::basic_string<char> >&) [with T = long long unsigned int]'
/ws/istio-master/envoy/source/server/options_impl.cc:120:1:   required from here
/ws/istio-master/src/tclap/include/tclap/ArgTraits.h:80:39: error: 'long long unsigned int' is not a class, struct, or union type
     typedef typename T::ValueCategory ValueCategory;

Comment (costinm) [reply source/server/options_impl.cc]: Also - I don't think any of the options would need 64 bit parameters.

Comment (costinm) [reply source/server/options_impl.cc]: That's with arm-rpi-4.9.3-linux-gnueabihf

Comment (htuch) [reply source/server/options_impl.cc]: Might be an issue in converting from the literal to the type. I think in general, for clean 32/64-bit support, we should have some macros for writing fixed width literals, e.g. `CONST64(0)`, since we're going to hit this elswhere. I do agree that these are fine as `uint32_t` though.

**Comment by mattklein123** [comment]
In general this seems fine to me but please fix DCO and format.

File: include/envoy/http/codec.h [hunk 2]
```diff
@@ -197,7 +197,7 @@ struct Http2Settings {
   // initial value from HTTP/2 spec, same as NGHTTP2_DEFAULT_HEADER_TABLE_SIZE from nghttp2
   static const uint32_t DEFAULT_HPACK_TABLE_SIZE = (1 << 12);
   // no maximum from HTTP/2 spec, use unsigned 32-bit maximum
-  static const uint32_t MAX_HPACK_TABLE_SIZE = (1UL << 32) - 1;
+  static const uint32_t MAX_HPACK_TABLE_SIZE = 0xFFFFFFFF;
```
Comment (dnoe) [inline include/envoy/http/codec.h]: You should be able to use the C++ numeric_limits for this:

```
#include <limits>
static const uint32_t MAX_HPACK_TABLE_SIZE = std::numeric_limits<uint32_t>::max();
```

File: source/server/options_impl.cc (hunk 1 above)
Comment (costinm) [reply source/server/options_impl.cc]: Seems specific to tclap macros - so far everything else seems to work on 32 bit arm, I'll do more testing and try to get it into a continuous build.

File: include/envoy/http/codec.h (hunk 2 above)
Comment (costinm) [reply include/envoy/http/codec.h]: Done

**Comment by costinm** [comment]
Git mixups attempting to edit commits to add DCO - will upload a clean PR
